src/components/Pages/RequestPage/RequestPage.tsx

```diff
@@ -65,6 +65,19 @@ export const RequestPage = () => {
         // Try to restablish connection with the wallet.
         const connectionData = await connection.tryPreviousConnection()
         providerRef.current = new ethers.BrowserProvider(connectionData.provider)
+
+        // Goes to the login page if no account is returned in the data.
+        if (!connectionData.account) {
+          throw new Error('No account connected')
+        }
+
+        const profile = await fetchProfile(connectionData.account)
```

Comment:
The profile is fetched as well on the setup page, I'd like to do something to prevent multiple fetches on another PR